avocado/plugins/replay.py

```diff
@@ -0,0 +1,96 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat, Inc. 2020
+# Authors: Cleber Rosa <crosa@redhat.com>
+
+"""Replay Job Plugin"""
+
+import json
+import os
+import sys
+
+from avocado.core.data_dir import get_job_results_dir
+from avocado.core.dispatcher import JobPrePostDispatcher
+from avocado.core.future.settings import settings
+from avocado.core import job, exit_codes, output
+from avocado.core.plugin_interfaces import CLICmd
+
+
+class Replay(CLICmd):
+
+    """Implements the avocado 'replay' subcommand."""
+
+    name = 'replay'
+    description = "Runs a new job using a previous job as its configuration"
+
+
+    def configure(self, parser):
+        parser = super(Replay, self).configure(parser)
+
+        help_msg = ('Replays a job, identified by: complete or partial Job '
+                    'ID, "latest" for the latest job, the job results path.')
+        settings.register_option(section='job.replay',
+                                 key='source_job_id',
+                                 help_msg=help_msg,
+                                 metavar='SOURCE JOB ID',
+                                 default='latest',
+                                 nargs='?',
+                                 positional_arg=True,
+                                 parser=parser)
+
+    @staticmethod
+    def _get_source_job_results_dir(source_job_id):
+        results_dir = get_job_results_dir(source_job_id)
+        if results_dir:
+            return results_dir
+        msg = 'Could not find the results directory for Job "%s"'
+        output.LOG_UI.error(msg, source_job_id)
+        sys.exit(exit_codes.AVOCADO_FAIL)
```

Comment:
This method name is "_get_..." but it is aborting in case of failure. I would recommend to raise a exception here and catch that and exit at the caller.

Reply:
I tried a similar approach, and the code was a lot less straightforward.  I'll try it again with the last 3 lines (log error / exit) as a utility method because that's a repeating pattern.  But, I'll also consider renaming the method.